spec/rspec/matchers/built_in/yield_spec.rb

```diff
@@ -628,6 +695,15 @@ def each_arg(*args, &block)
         |         got: ["food", "barn"]
       EOS
     end
+
+    it 'fails when the successively yielded args match the matchers at return time only' do
+      expect { |b|
+        %w[ barn food ].each do |val|
+          _yield_with_args(val, &b)
+          val.delete!(val)
+        end
+      }.not_to yield_successive_args(a_string_matching(/foo/), a_string_matching(/bar/))
+    end
```

Comment:
> I _think_ that's right

I independently tried getting this test failing on master and passing on this branch and came up with the same thing, so that's a good sign!